vernac/search.ml

```diff
@@ -153,18 +153,12 @@ end
 module PriorityQueue = Heap.Functional(ConstrPriority)
 
 let rec iter_priority_queue q fn =
-  (* use an option to make the function tail recursive. Will be
-     obsoleted with Ocaml 4.02 with the [match … with | exception …]
-     syntax. *)
-  let next = begin
-      try Some (PriorityQueue.maximum q)
-      with Heap.EmptyHeap -> None
-  end in
-  match next with
-  | Some (gref,kind,env,t,_) ->
+  (* Tail-rec! *)
```

Comment:
We could use the tailrec attribute instead of an unchecked comment.

Reply:
I didn't manage to make this work actually, and it's probably not very important either.